### ISSUE TITLE
The intermediate output for stream (CTR) is not handled properly

### ISSUE THREAD

**Issue by rhishikeshj** [krzyzanowskim/CryptoSwift#635]
- CryptoSwift version: 0.9.0
- Swift version: 4
- Platform (iOS/macOS/Linux/...): iOS 11.4
- Installation method (CocoaPods/Carthage/SPM/...): Cocoapods
- Project name: SampleApp

# Description
In the AES/CTR mode with noPadding, I am getting different results when encrypting some data multiple times.
My reference implementation is Python which seems to agree with the server side code written in C and also the Java implementation which I use in Android.
But the implementation in CryptoSwift only seems to agree with the above reference implementations for the first 3 times !
For the fourth time I call encrypt, it starts giving different results.
Please find below the Swift code as well as Python implementations.

> To whom may be concerned: I offer professional support to all my open source projects.

# Steps to reproduce
Run the below code in Swift and then the Python code.
Values should match.

Sample code to reproduce the issue:

```swift
       import CryptoSwift

// Code goes here
        let sharedKey = Data(hex: "ff72161a98db39eb81b92079e937120b67fefe3f07d53ebf995d5422eb4a7ee4")
        let iv = Data(hex: "54b37c9f12934210bc44f01b2752dbce")
        var fencryptor = try! AES(key: sharedKey.bytes,
                              blockMode: BlockMode.CTR(iv: iv.bytes),
                              padding: .noPadding).makeEncryptor()

        let devicePublicKey = Data(hex: "5e10a67b08a33dcd4d8fb3edda5c4cd7f11b14a5da0ca96972f21cdba2bf0444")
        let deviceVerifyData = Data(hex: "b9b808c634aab0387f6cc02ee30e1e606c6d6d4dcb9a18fa1d9fc94a1cb267bd")
        
        let encryptedPK = try! fencryptor.update(withBytes: devicePublicKey.bytes, isLast: true)
        let decryptedVD = try! fencryptor.update(withBytes: deviceVerifyData.bytes, isLast: true)
        print("Encrypted pk : \(encryptedPK.toHexString())")
        print("Decrypted device verify : \(decryptedVD.toHexString())")
        
        let wifiConfigRequest = Data(hex: "5219080262150a0954656a6f6e6964686912085a6f657932363131")
        let encryptedRequest = try! fencryptor.update(withBytes: wifiConfigRequest.bytes, isLast: true)
        print("Encrypted request : \(encryptedRequest.toHexString())")

        let response = Data(hex: "083a536260fb")
        let decryptedResponse = try! fencryptor.update(withBytes: response.bytes, isLast: true)
        print("Decrypted response : \(decryptedResponse.toHexString())")
        
        let data = Data(hex: "0123456789abcdef")
        print("Encrypted sample data \(try! fencryptor.update(withBytes: data.bytes, isLast: true).toHexString())")
        print("Decrypted sample data \(try! fencryptor.update(withBytes: data.bytes, isLast: true).toHexString())")

```

Python code

```python
import curve25519
import Crypto.Hash.SHA256
import Crypto.Cipher.AES
import Crypto.Util.Counter

devicePublicKey = '5e10a67b08a33dcd4d8fb3edda5c4cd7f11b14a5da0ca96972f21cdba2bf0444'.decode('hex')
clientPublicKey = '6639a00b5b9c158c7317eef5f981dc4feee7ed42a2d7098a80937e4dcada1057'.decode('hex')

sharedK = 'ff72161a98db39eb81b92079e937120b67fefe3f07d53ebf995d5422eb4a7ee4'.decode('hex')
deviceRandom = '54b37c9f12934210bc44f01b2752dbce'.decode('hex')

ctr = Crypto.Util.Counter.new(128, initial_value=long(deviceRandom.encode('hex'), 16))
encipher = Crypto.Cipher.AES.new(sharedK, Crypto.Cipher.AES.MODE_CTR, counter=ctr)

deviceVerifyData = 'b9b808c634aab0387f6cc02ee30e1e606c6d6d4dcb9a18fa1d9fc94a1cb267bd'.decode('hex')

print('Device public key encrypted : ', encipher.encrypt(devicePublicKey).encode('hex'))

deviceVerifyData = encipher.decrypt(deviceVerifyData)

print('Decrypted client public key is ', deviceVerifyData.encode('hex'))

setWifiConfigRequest = '5219080262150a0954656a6f6e6964686912085a6f657932363131'.decode('hex')
encryptedConfigRequest = encipher.encrypt(setWifiConfigRequest)
print('Encrypted request : ', encryptedConfigRequest.encode('hex'))

response = '083a536260fb'.decode('hex')
decryptedResponse = encipher.decrypt(response)
print('Decrypted response :', decryptedResponse.encode('hex'))

data = '0123456789abcdef'.decode('hex')
print('Encrypted sample data : ', encipher.encrypt(data).encode('hex'))
print('Decrypted sample data : ', encipher.decrypt(data).encode('hex'))

```

Swift output

```
Encrypted pk : b74b1bfbfae07c993d6d61017cbdf7d4c9e1c021ab14df0765cf260d845af56c
Decrypted device verify : 333bdc72b8c1f6026ca682451c0f781a0b705dfb20f1cae8c3749f68f2f0dfbe
Encrypted request : a84dfb87674af818cc7f4459def18b9db90d205202ac11eef07192
Decrypted response : f3ba09440147
Encrypted sample data d4d962d94acee4c2
Decrypted sample data eb3252c933b0528a

```

Python output
```
('Device public key encrypted : ', 'b74b1bfbfae07c993d6d61017cbdf7d4c9e1c021ab14df0765cf260d845af56c')
('Decrypted client public key is ', '333bdc72b8c1f6026ca682451c0f781a0b705dfb20f1cae8c3749f68f2f0dfbe')
('Encrypted request : ', 'a84dfb87674af818cc7f4459def18b9db90d205202ac11eef07192')
('Decrypted response :', '520408036a00')
('Encrypted sample data : ', '8179630635d35cb2')
('Decrypted sample data : ', 'd94aac64b419223a')
```

**Comment by krzyzanowskim** [comment]
Looks like the counter increases too early. That would work:

```swift
let wifiConfigRequest = Data(hex: "5219080262150a0954656a6f6e6964686912085a6f657932363131083a536260fb")
```
Encrypted request : a84dfb87674af818cc7f4459def18b9db90d205202ac11eef07192**520408036a00**

Related issue with stream block modes https://github.com/krzyzanowskim/CryptoSwift/issues/397

**Comment by krzyzanowskim** [comment]
Proper stream support for CTR mode landed in 0.11.0. Closing.
